plugins/generator-1.18.2/forge-1.18.2/templates/elementinits/villagerprofessions.java.ftl

```diff
@@ -0,0 +1,86 @@
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # Additional permission for code generator templates (*.ftl files)
+ #
+ # As a special exception, you may create a larger work that contains part or
+ # all of the MCreator code generator templates (*.ftl files) and distribute
+ # that work under terms of your choice, so long as that work isn't itself a
+ # template for code generation. Alternatively, if you modify or redistribute
+ # the template itself, you may (at your option) remove this special exception,
+ # which will cause the template and the resulting code generator output files
+ # to be licensed under the GNU General Public License without this special
+ # exception.
+-->
+
+<#-- @formatter:off -->
+
+<#include "../mcitems.ftl">
+
+/*
+*    MCreator note: This file will be REGENERATED on each build.
+*/
+
+package ${package}.init;
+
+import net.minecraft.sounds.SoundEvent;
+
+public class ${JavaModName}VillagerProfessions {
+
+    public static final DeferredRegister<PoiType> POI = DeferredRegister.create(ForgeRegistries.POI_TYPES, ${JavaModName}.MODID);
+    public static final DeferredRegister<VillagerProfession> PROFESSIONS = DeferredRegister.create(ForgeRegistries.PROFESSIONS, ${JavaModName}.MODID);
+
+    <#list villagerprofessions as villagerprofession>
+        public static final RegistryObject<VillagerProfession> ${villagerprofession.getModElement().getRegistryNameUpper()} =
+            registerProfession(
+			    "${villagerprofession.getModElement().getRegistryName()}",
+                ${mappedBlockToBlock(villagerprofession.pointOfInterest)},
+                ImmutableSet.of(),
+                ImmutableSet.of(),
+                () -> new SoundEvent(new ResourceLocation("${villagerprofession.actionSound}"))
+            );
+    </#list>
+
+	private static RegistryObject<VillagerProfession> registerProfession(String name, Block block, ImmutableSet<Item> specificItems, ImmutableSet<Block> relatedWorldBlocks, Supplier<SoundEvent> soundEventSupplier) {
+		Optional<PoiType> existingCheck = PoiType.forState(block.defaultBlockState());
+
+		if (existingCheck.isPresent()) {
+            ${JavaModName}.LOGGER.error("Skipping villager profession " + name + " that uses POI block " + block + " that is already in use by " + existingCheck);
+			return null;
+		}
+
+		Supplier<PoiType> poi = POI.register(name, () -> new PoiType(name, ImmutableSet.copyOf(block.getStateDefinition().getPossibleStates()), 1, 1));
+        return PROFESSIONS.register(name, () -> new RegistrySafeVillagerProfession(${JavaModName}.MODID + ":" + name, poi.get(), specificItems, relatedWorldBlocks, soundEventSupplier));
+    }
+
+    public static class RegistrySafeVillagerProfession extends VillagerProfession {
```

Comment:
Oh, and I just noticed it: given the `RegistrySafeVillagerProfession` class being generated for 1.18.2 is not used on version 1.19.2, can't it be inlined for 1.18.2 too? (Question to Klemen, ~keep out~)

Reply:
The issue is with the work sound, in 1.18.2 it needs to be added through a class that extends the `VillagerProfession` class with an `@Override`

Reply:
Why can't you just replace `Supplier<SoundEvent>` with just `SoundEvent` for 118.2 if this is done in 1.19.2?

Reply:
In 1.19.2 it seems to work without this workaround (which is also not possible due to class now being a record)